pgmpy/models/MarkovModel.py

```diff
@@ -676,3 +676,30 @@ def get_partition_function(self):
             raise ValueError('Factor for all the random variables not defined.')
 
         return np.sum(factor.values)
+
+    def copy(self):
+        """
+        Returns a deep copy of this Markov Model.
+ 
+        Returns
+        -------
+        MarkovModel: Deep copy of this Markov model.
+ 
+        Examples
+        -------
+        >>> G = MarkovModel()
+        >>> G.add_nodes_from([('a', 'b'), ('b', 'c')])
+        >>> G.add_edge(('a', 'b'), ('b', 'c'))
+        >>> G_copy = G.copy()
+        >>> G_copy.edges()
+        [(('a', 'b'), ('b', 'c'))]
+        >>> G_copy.nodes()
+        [('a', 'b'), ('b', 'c')]
+        """
+        clone_graph = MarkovModel(self.edges())
+
```

Comment:
@liquidmetal You must add all the nodes, in the graph. There might be some nodes which don't have any edges.